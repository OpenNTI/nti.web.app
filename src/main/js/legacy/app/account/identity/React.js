const Ext = require('@nti/extjs');
const { getService } = require('@nti/web-client');
const { Identity } = require('@nti/web-profiles');
const LoginActions = require('internal/legacy/login/Actions');
require('internal/legacy/overrides/ReactHarness');

const AccountActions = require('../Actions');

module.exports = exports = Ext.define(
	'NextThought.app.account.identity.React',
	{
		extend: 'NextThought.ReactHarness',
		alias: 'widget.identity-react',
		component: Identity.Session,

		setTheme(scope, knockout) {},

		initComponent: function () {
			this.callParent(arguments);
			Object.assign(this.initialConfig, {
				addHistory: true,
				baseroute: '/app',
			});
			this.AccountActions = AccountActions.create();
			this.LoginActions = LoginActions.create();
		},

		async afterRender() {
			this.callParent(arguments);
			const service = await getService();
			const supportLinks = service.getSupportLinks();

			this.mon(Ext.getBody(), {
				click: this.onBodyClicks.bind(this, supportLinks),
			});
		},

		onBodyClicks(supportLinks, e) {
			let action = e.getTarget('a[data-action]')?.dataset.action;
			const link = e.getTarget('a[href]');
			const href = link?.getAttribute('href');

			// This is used to launch support logic the same way everywhere
			// we have a link that goes to the supportContact url. (so we can
			// just emit that url in the dom and not worry about coping logic)
			if (
				supportLinks.supportContact === href &&
				supportLinks.internalSupport
			) {
				action = action || 'support';
			}

			if (action) {
				if (call(this, `do-${action}`, link, e) !== false) {
					e.stopEvent();
				}
			}
		},

		showLink(target) {
			target.setAttribute('target', '_blank');
			return false;
		},

		'do-welcomeGuide'(target) {
			this.AccountActions.showWelcomePage(target.href);
		},

		'do-about': 'showLink',

		'do-privacy'(target) {
			this.AccountActions.showPrivacy(target);
		},

		'do-privacyForMinors'(target) {
			this.AccountActions.showChildrensPrivacy(target.href);
		},

		'do-terms'(target) {
			this.AccountActions.showTermsOfService(target.href);
		},

		'do-support'(target) {
			this.AccountActions.showContactUs();
		},

		'do-helpSite': 'showLink',

		'do-impersonate'() {
			this.LoginActions.handleImpersonate();
		},

		'do-logout'() {
			this.LoginActions.handleLogout();
		},
	}
);

function call(scope, action, ...args) {
	if (typeof action === 'string') {
		return call(scope, scope[action], ...args);
	}

	return action.apply(scope, args);
}
