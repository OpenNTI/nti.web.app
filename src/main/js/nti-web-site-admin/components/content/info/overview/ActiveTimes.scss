.site-admin-course-overview {
	.active-times-widget {
		margin-left: 0;
	}
}
