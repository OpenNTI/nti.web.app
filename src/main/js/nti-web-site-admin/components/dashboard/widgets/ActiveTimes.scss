@import '~@nti/style-common';

.active-times-widget {
	height: 650px;
	padding: 25px;
	box-shadow: 0 1px 2px 1px #ccc;
	color: #555;

	.active-times-header {
		font-size: 24px;
		margin-bottom: 30px;
	}

	.active-times-chart {
		height: 455px;

		.active-blocks {
			display: flex;

			.active-block {
				width: 48px;
				height: 24px;
				background-color: #eee;
				margin-right: 10px;
				margin-bottom: 10px;
			}

			.time-label {
				width: 48px;
				height: 24px;
				line-height: 24px;
				font-size: 12px;
				text-align: right;
				font-weight: 600;
				text-transform: uppercase;
				color: var(--secondary-grey);
			}
		}

		.days-of-week {
			display: flex;

			.day {
				width: 48px;
				height: 24px;
				margin-right: 10px;
				margin-bottom: 10px;
				font-size: 12px;
				text-align: center;
				text-transform: uppercase;
				font-weight: 600;
				color: var(--secondary-grey);
			}
		}
	}

	.active-times-popular {
		border-top: solid 1px #ccc;

		.active-times-popular-header {
			color: #999;
			margin-top: 20px;
			margin-bottom: 4px;
		}

		.active-times-popular-content {
			font-size: 32px;
		}
	}
}
