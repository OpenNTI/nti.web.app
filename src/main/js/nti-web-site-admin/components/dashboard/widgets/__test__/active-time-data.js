export const mockActiveTimeData = {
	WeekDays: {
		Friday: [0, 0, 4, 7, 4, 2, 8, 5, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
		Monday: [3, 5, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3, 6, 0, 0, 0, 15, 5, 2, 1, 0, 0],
		Saturday: [4, 6, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
		Sunday: [10, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
		Thursday: [9, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 5, 0, 2, 1, 0, 0, 6, 0, 0],
		Tuesday: [8, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
		Wednesday: [5, 6, 7, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 6, 3, 0, 0, 2, 0, 0, 0, 0, 0]
	}
};
