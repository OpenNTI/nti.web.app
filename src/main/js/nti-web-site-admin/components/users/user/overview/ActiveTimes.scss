.site-admin-user-overview {
	.active-times-widget {
		margin: 0;
	}
}
