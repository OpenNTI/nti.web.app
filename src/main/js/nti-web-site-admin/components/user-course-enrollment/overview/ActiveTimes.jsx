import React from 'react';
import PropTypes from 'prop-types';
import { getService } from '@nti/web-client';
import { Widgets } from '@nti/web-reports';

const ANALYTICS_LINK = 'analytics';
const ACTIVE_TIMES_SUMMARY_LINK = 'active_times_summary';
const { ActiveTimesChart } = Widgets;

export default class ActiveTimes extends React.Component {
	static propTypes = {
		enrollment: PropTypes.object.isRequired,
	};

	constructor(props) {
		super(props);

		this.state = {};
	}

	componentDidMount() {
		this.loadData();
	}

	async loadData() {
		const { enrollment } = this.props;

		const service = await getService();
		const analyticsLink = enrollment.getLink(ANALYTICS_LINK);
		const results = (await service.getBatch(analyticsLink)) || {};
		const activeTimesSummaryLink = results.getLink(
			ACTIVE_TIMES_SUMMARY_LINK
		);
		const summaryData = (await service.get(activeTimesSummaryLink)) || {};

		this.setState({ data: summaryData.WeekDays });
	}

	render() {
		return (
			<ActiveTimesChart data={this.state.data} error={this.state.error} />
		);
	}
}
