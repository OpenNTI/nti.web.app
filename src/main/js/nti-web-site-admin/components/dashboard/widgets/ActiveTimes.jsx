import './ActiveTimes.scss';
import React from 'react';
import { getService } from '@nti/web-client';
import { getLink } from '@nti/lib-interfaces';
import { Widgets } from '@nti/web-reports';

const { ActiveTimesChart } = Widgets;

const ANALYTICS = 'Analytics';
const ACTIVE_TIMES_LINK = 'active_times_summary';

export default class ActiveTimes extends React.Component {
	constructor(props) {
		super(props);

		this.state = {};
	}

	componentDidMount() {
		this.loadData();
	}

	async loadData() {
		try {
			const service = await getService();
			const sessionsCollection = service.getWorkspace(ANALYTICS);
			const link = getLink(sessionsCollection, ACTIVE_TIMES_LINK);
			const stats = await service.get(link);
			this.setState({ data: stats['WeekDays'] });
		} catch (e) {
			this.setState({ error: e });
		}
	}

	render() {
		return (
			<ActiveTimesChart data={this.state.data} error={this.state.error} />
		);
	}
}
