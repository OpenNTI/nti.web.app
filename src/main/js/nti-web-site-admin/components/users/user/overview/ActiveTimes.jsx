import './ActiveTimes.scss';
import React from 'react';
import PropTypes from 'prop-types';
import {getService} from '@nti/web-client';
import {Widgets} from '@nti/web-reports';

const ANALYTICS_LINK = 'analytics';
const ACTIVE_TIMES_SUMMARY_LINK = 'active_times_summary';

const { ActiveTimesChart } = Widgets;

export default class ActiveTimes extends React.Component {
	static propTypes = {
		user: PropTypes.object.isRequired
	}

	constructor (props) {
		super(props);

		this.state = {};
	}

	componentDidMount () {
		this.loadData();
	}

	async loadData () {
		try {
			const { user } = this.props;

			const service = await getService();
			const analyticsLink = user.getLink(ANALYTICS_LINK);
			const results = await service.getBatch(analyticsLink) || {};
			const activeTimesSummaryLink = results.getLink(ACTIVE_TIMES_SUMMARY_LINK);
			const summaryData = await service.get(activeTimesSummaryLink) || {};

			this.setState({data: summaryData.WeekDays});
		}
		catch (e) {
			this.setState({error: e});
		}
	}

	render () {
		return (<ActiveTimesChart data={this.state.data} error={this.state.error}/>);
	}
}
