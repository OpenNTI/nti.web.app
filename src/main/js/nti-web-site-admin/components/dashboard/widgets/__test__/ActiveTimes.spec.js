import React from 'react';
import renderer from 'react-test-renderer';
import { TestUtils } from '@nti/web-client';
import { Date as DateUtils } from '@nti/lib-commons';

import ActiveTimes from '../ActiveTimes';

import {mockActiveTimeData} from './active-time-data';

const { tearDownTestClient, setupTestClient } = TestUtils;
const { MockDate } = DateUtils;

const getMockService = () => {
	return {
		getCollection: () => {
			return {
				hasLink: () => true,
				getLink: () => 'mockLink'
			};
		},
		getWorkspace: () => {
			return {};
		},
		get: () => {
			return Promise.resolve(mockActiveTimeData);
		}
	};
};

const onBefore = () => {
	jest.useFakeTimers();
	setupTestClient(getMockService());
};

const onAfter = () => {
	tearDownTestClient();
};

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

/* eslint-env jest */
describe('Site admin dashboard widget active times', () => {
	const actingDate = new Date('10/31/2018');
	MockDate.install(actingDate);

	beforeEach(() => onBefore());
	afterEach(onAfter);

	afterAll(() => {
		MockDate.uninstall();
	});

	test('Basic render test', async () => {
		const cmp = renderer.create(<ActiveTimes/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
