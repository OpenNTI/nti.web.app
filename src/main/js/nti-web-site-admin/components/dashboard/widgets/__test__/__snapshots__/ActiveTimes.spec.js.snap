// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin dashboard widget active times Basic render test 1`] = `
<div
  className="active-times-widget"
>
  <div
    className="active-times-header"
  >
    Active Times
  </div>
  <div
    className="active-times-chart"
  >
    <div
      className="active-blocks"
    >
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#3182bd",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#6baed6",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#6baed6",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#3182bd",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#3182bd",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#6baed6",
          }
        }
      />
      <div
        className="time-label"
      >
        12 AM
      </div>
    </div>
    <div
      className="active-blocks"
    >
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#9ecae1",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#3182bd",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="time-label"
      >
        2 AM
      </div>
    </div>
    <div
      className="active-blocks"
    >
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#9ecae1",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="time-label"
      >
        4 AM
      </div>
    </div>
    <div
      className="active-blocks"
    >
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#3182bd",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="time-label"
      >
        6 AM
      </div>
    </div>
    <div
      className="active-blocks"
    >
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="time-label"
      >
        8 AM
      </div>
    </div>
    <div
      className="active-blocks"
    >
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="time-label"
      >
        10 AM
      </div>
    </div>
    <div
      className="active-blocks"
    >
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#dfdfef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#dfdfef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="time-label"
      >
        12 PM
      </div>
    </div>
    <div
      className="active-blocks"
    >
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#9ecae1",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#6baed6",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#6baed6",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="time-label"
      >
        2 PM
      </div>
    </div>
    <div
      className="active-blocks"
    >
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#dfdfef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="time-label"
      >
        4 PM
      </div>
    </div>
    <div
      className="active-blocks"
    >
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#08306b",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#dfdfef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#dfdfef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="time-label"
      >
        6 PM
      </div>
    </div>
    <div
      className="active-blocks"
    >
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#dfdfef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#9ecae1",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="time-label"
      >
        8 PM
      </div>
    </div>
    <div
      className="active-blocks"
    >
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="active-block"
        style={
          Object {
            "backgroundColor": "#efefef",
          }
        }
      />
      <div
        className="time-label"
      >
        10 PM
      </div>
    </div>
    <div
      className="days-of-week"
    >
      <div
        className="day"
      >
        Sun
      </div>
      <div
        className="day"
      >
        Mon
      </div>
      <div
        className="day"
      >
        Tues
      </div>
      <div
        className="day"
      >
        Wed
      </div>
      <div
        className="day"
      >
        Thur
      </div>
      <div
        className="day"
      >
        Fri
      </div>
      <div
        className="day"
      >
        Sat
      </div>
    </div>
  </div>
  <div
    className="active-times-popular"
  >
    <div
      className="active-times-popular-header"
    >
      Most Popular Time
    </div>
    <div
      className="active-times-popular-content"
    >
      Monday
      , 
      6 PM
       – 
      8 PM
    </div>
  </div>
</div>
`;
